subsys/bluetooth/audio/bap_broadcast_assistant.c

```diff
@@ -381,11 +383,18 @@ static uint8_t notify_handler(struct bt_conn *conn,
 		} else {
 			return parse_and_send_recv_state(conn, handle, data, length, &recv_state);
 		}
-	} else if (broadcast_assistant_cbs != NULL &&
-		   broadcast_assistant_cbs->recv_state_removed != NULL) {
-		broadcast_assistant.past_avail[index] = false;
-		broadcast_assistant_cbs->recv_state_removed(conn, 0,
-							    broadcast_assistant.src_ids[index]);
+	} else {
+		struct bt_bap_broadcast_assistant_cb *listener, *next;
+
+		SYS_SLIST_FOR_EACH_CONTAINER_SAFE(&broadcast_assistant_cbs,
+						  listener, next, _node) {
+			if (listener->recv_state_removed) {
+				broadcast_assistant.past_avail[index] = false;
```

Comment:
This looks like a bug: `broadcast_assistant.past_avail[index] = false;` is only done if there is a callback, but it should always be done, right? Suggest to fix in this PR, but as a separate commit